metaflow/R.py

```diff
@@ -12,6 +11,15 @@
 R_VERSION = None
 R_VERSION_CODE = None
 
+try:
+    def source(fullname, path):
+        from importlib.machinery import SourceFileLoader
+        return SourceFileLoader(fullname, path).load_module()
```

Comment:
Since this is to support things going forward, this seems deprecated in 3.6 (https://docs.python.org/3/library/importlib.html#importlib.machinery.SourceFileLoader.load_module) and we should now use: https://docs.python.org/3/library/importlib.html#importlib.abc.Loader.exec_module. I would try with exec_module first (already present in 3.4 and so most likely works), and if that doesn't work, fall back to imp (or maybe fallback to load_module and then imp if you want to be fancy but I don't think that is worth the trouble).

Reply:
Good call. I'll push up the change and let the CI confirm that it works.

Reply:
The tests are unhappy. Looks like `SourceFileLoader.exec_module` has a required argument (`module`) that I neglected to include in the most recent commit. It's not immediately clear to me what that argument should be, so I think I'll have to read the `importlib` docs more closely before I can fix.

Reply:
Hmm, I extracted out the 45f915d version of `source`, and it seems to be doing what it's supposed to:
```
$ cat foo.py
#!/usr/bin/env python

import importlib.machinery
import importlib.util

def source(fullname, path):
    loader = importlib.machinery.SourceFileLoader(fullname, path)
    spec = importlib.util.spec_from_loader(loader.name, loader)
    module = importlib.util.module_from_spec(spec)
    loader.exec_module(module)
    return module

module = source("bar", "bar.py")
print(module)
module.hello()

$ cat bar.py 
def hello():
    print("hello")

$ ./foo.py 
<module 'bar' from 'bar.py'>
hello
```
But the tests are failing with:
```
 Error in py_call_impl(callable, dots$args, dots$keywords) : 
  ModuleNotFoundError: No module named 'metaflowR'
```